worker/src/main.rs

```diff
@@ -113,6 +113,7 @@ fn main() {
         worker_accessor.clone(),
     ));
     let node_api_factory = Arc::new(NodeApiFactoryImpl::new(config.node_url()));
+    write_node_url(config.node_url());
```

Comment:
The `new` method of `NodeApiFactoryImpl` already writes the node url - why do it a second time?

Reply:
Ah, this is an error, I wanted to do the poc first without a factory at all. Then I changed my mind and forgot to remove that one.